spec/unit/util/plist_spec.rb

```diff
@@ -132,6 +132,13 @@
       expect(subject.parse_plist(valid_xml_plist)).to eq(valid_xml_plist_hash)
     end
 
+    it "forces UTF-8 encoding on an ASCII XML plist" do
+      ascii_xml_plist = '<?xml version="1.0" encoding="UTF-8"?>'.encode(Encoding::US_ASCII)
+      expect(ascii_xml_plist).to receive(:force_encoding).with(Encoding::UTF_8)
+
+      subject.parse_plist(ascii_xml_plist)
+    end
```

Comment:
I try to avoid writing tests that verify methods are called as that's an implementation detail. I find it better to test what the method returns. So in this case verifying that the returned string is UTF-8 encoded and that the contents of the string are what we expect.

It sounds like the failure occurs because the `dscl -plist . readall /Users` command returns a string whose `String#encoding` method returns `US-ASCII`, but the bytes are not actually US-ASCII? If so, are the bytes UTF-8 or UTF-16BE (it seems [cfpropertylist supports both](https://github.com/ckruse/CFPropertyList/blob/master/lib/cfpropertylist/rbCFTypes.rb#L47)).

It would be good to pass whatever data is making puppet fail into the `parse_plist` method and verify it returns the expected thing, something like:

```ruby
xml = <<XML
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
	<dict>
		<key>dsAttrTypeStandard:RecordName</key>
		<array>
			<string>Timișoara</string>
		</array>
	</dict>
</array>
</plist>
XML

notascii_xml_plist = xml.encode(Encoding::UTF_8).force_encoding(Encoding::US_ASCII)
plist = subject.parse_plist(notascii_xml_plist)
expect(plist.class.to_s).to eq(::CFDictionary)
expect(etc)
```

Reply:
> If so, are the bytes UTF-8 or UTF-16BE

Based on the doctype information it's UTF-8, and Puppet fails when doing the `plist_data =~ bad_xml_doctype` match, so we don't even get to creating a new CFPropertyList.